src/main/haskell/kore/src/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -705,27 +787,48 @@ variableFunctionAndEquals
 variableFunctionAndEquals
     simplificationType
     tools
+    substitutionSimplifier
+    (PredicateSubstitutionMerger substitutionMerger)
     (Var_ v)
     second
-  | isFunctionPattern tools second =
+  | isFunctionPattern tools second = Monad.Trans.lift $ do -- ExceptT Simplifier
+    Predicated {term = (), predicate, substitution} <- Monad.Trans.lift $
+        case simplificationType of -- Simplifier
+            SimplificationType.And ->
+                -- Ceil predicate not needed since 'second' being bottom
+                -- will make the entire term bottom. However, one must
+                -- be careful to not just drop the term.
+                return ExpandedPattern.topPredicate
+            SimplificationType.Equals -> do
+                (resultOr, _proof) <-
+                    Ceil.makeEvaluateTerm tools substitutionSimplifier second
+                case OrOfExpandedPattern.extractPatterns resultOr of
+                    [] -> return ExpandedPattern.bottomPredicate
+                    [resultPredicateSubstitution] ->
+                        return resultPredicateSubstitution
+                    _ -> error
```

Comment:
could you please add a comment / example here on when/ why this could be happening.